client/client.go

```diff
@@ -91,17 +106,34 @@ func NewClient(config *Config) (*Client, error) {
 		return nil, fmt.Errorf("could not parse APIUrl: %w", err)
 	}
 
-	httpClient := &http.Client{}
-	if cfg.Debug {
-		httpClient = httputil.WrapWithLogging(httpClient)
-	}
-
 	client := &Client{
-		apiKey:     cfg.APIKey,
-		apiURL:     apiURL,
-		userAgent:  cfg.UserAgent,
-		httpClient: httpClient,
+		apiKey:  cfg.APIKey,
+		apiURL:  apiURL,
+		headers: make(http.Header),
+	}
+
+	client.httpClient = &retryablehttp.Client{
+		Backoff:      retryablehttp.DefaultBackoff,
+		CheckRetry:   client.retryHTTPCheck,
+		ErrorHandler: retryablehttp.PassthroughErrorHandler,
+		HTTPClient:   cfg.HTTPClient,
+		RetryWaitMin: 200 * time.Millisecond,
+		RetryWaitMax: 10 * time.Second,
+		RetryMax:     15,
```

Comment:
Would we want to make these values configurable somehow?

Reply:
I don't _think_ so at this point but I thought about it as well. If this were a generic client then I think yes, but I think/hope these should be "good enough" (for now, anyway)